wallet/src/main/scala/org/bitcoins/wallet/models/AddressTagTable.scala

```diff
@@ -0,0 +1,48 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.wallet.utxo.{AddressTag, AddressTagType}
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.{PrimaryKey, ProvenShape}
+
+case class AddressTagDb(
+    address: BitcoinAddress,
+    tag: AddressTag,
+    tagType: AddressTagType)
+
+class AddressTagTable(t: Tag)
+    extends Table[AddressTagDb](t, "wallet_address_tags") {
+
+  import org.bitcoins.db.DbCommonsColumnMappers._
+
+  def address: Rep[BitcoinAddress] = column[BitcoinAddress]("address")
+
+  def tag: Rep[AddressTag] = column[AddressTag]("tag")
+
+  def tagType: Rep[AddressTagType] = column[AddressTagType]("tagType")
+
+  private type AddressTagTuple = (BitcoinAddress, AddressTag, AddressTagType)
+
+  private val fromTuple: AddressTagTuple => AddressTagDb = {
+    case (address, tag, tagType) =>
+      AddressTagDb(address, tag, tagType)
+  }
+
+  private val toTuple: AddressTagDb => Option[AddressTagTuple] =
+    addrTag => Some((addrTag.address, addrTag.tag, addrTag.tagType))
+
+  override def * : ProvenShape[AddressTagDb] =
+    (address, tag, tagType) <> (fromTuple, toTuple)
+
+  /** All tags must have an associated address */
+  def fk_address = {
+    val addressTable = TableQuery[AddressTable]
+    foreignKey("fk_address",
+               sourceColumns = address,
+               targetTableQuery = addressTable)(_.address)
+  }
+
+  def primaryKey: PrimaryKey =
+    primaryKey("pk_address", sourceColumns = address)
```

Comment:
Doesn't an address show up many times if it has many tags? How can it be a primary key?